mlir/lib/CAPI/Dialect/MIGraphX.cpp

```diff
@@ -7,8 +8,126 @@
 //===----------------------------------------------------------------------===//
 
 #include "mlir-c/Dialect/MIGraphX.h"
+#include "mlir/CAPI/Pass.h"
 #include "mlir/CAPI/Registration.h"
+#include "mlir/Dialect/GPU/GPUDialect.h"
+#include "mlir/Dialect/GPU/Passes.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
 #include "mlir/Dialect/MIGraphX/MIGraphXOps.h"
+#include "mlir/Dialect/MIGraphX/Pipeline.h"
+#include "mlir/Dialect/MIOpen/MIOpen.h"
+#include "mlir/Dialect/MIOpen/Pipeline.h"
+#include "mlir/ExecutionEngine/OptUtils.h"
+#include "llvm/Support/TargetSelect.h"
+#include <vector>
 
 MLIR_DEFINE_CAPI_DIALECT_REGISTRATION(MIGraphX, migraphx,
                                       mlir::migraphx::MIGraphXDialect)
+
+// Returns the required buffer size if called with null buffer
+// and fill information in the passed ptr when provided.
+MLIR_CAPI_EXPORTED
+void mlirGetKernelInfo(MlirModule module, int *size, void *data) {
+  auto mod = unwrap(module);
+  int argNum = 0;
+  int argIdx = 0;
+  llvm::StringRef kernelName;
+  if (data == nullptr && size != nullptr) {
```

Comment:
Better to share the collection of info and just conditionally assign size and data.